src/commcare_cloud/fab/operations/supervisor.py

```diff
@@ -1,277 +1,20 @@
-
-
-@task
-@roles(ROLES_ALL_SERVICES)
-@parallel
-def set_supervisor_config():
-    """Upload and link Supervisor configuration from the template."""
-    set_celery_supervisorconf()
-    set_formsplayer_supervisorconf()
-    set_formplayer_spring_supervisorconf()
-
-
-def _get_celery_queues():
-    queues = {}
-    queues.update(env.celery_processes.get(env.get('host_string').split(':')[0], {}))
-
-    return queues
-
-
-def _check_in_roles(roles):
-    return any(env.get('host_string') in env.roledefs[role] for role in roles)
-
-
-def set_celery_supervisorconf():
-    if not _check_in_roles(ROLES_CELERY):
-        return
-
-    queues = _get_celery_queues()
-
-    if 'celery_periodic' in queues and env.host != queues['celery_periodic'].get('server_whitelist'):
-        show_periodic_server_whitelist_message_and_abort(env)
-
-    _rebuild_supervisor_conf_file('make_supervisor_conf', 'celery_bash_runner.sh')
-    _rebuild_supervisor_conf_file(
-        'make_supervisor_conf', 'celery_bash_runner.sh',
-        params={'python_options': '-O'}, conf_destination_filename='celery_bash_runner_optimized.sh'
-    )
-
-    for comma_separated_queue_names, params in queues.items():
-        queue_names = comma_separated_queue_names.split(',')
-        if queue_names == ['flower']:
-            _rebuild_supervisor_conf_file(
-                'make_supervisor_conf',
-                'supervisor_celery_flower.conf',
-                {'celery_params': params}
-            )
-            continue
-
-        pooling = params['pooling']
-        max_tasks_per_child = params['max_tasks_per_child']
-        num_workers = params.get('num_workers', 1)
-
-        params.update({
-            'comma_separated_queue_names': comma_separated_queue_names,
-            'pooling': pooling,
-            'max_tasks_per_child': max_tasks_per_child,
-        })
-
-        for worker_num in range(num_workers):
-            params.update({
-                'worker_num': worker_num,
-            })
-
-            conf_destination_filename = 'supervisor_celery_worker_{}_{}.conf'.format(
-                comma_separated_queue_names, worker_num)
-
-            worker_name = get_celery_worker_name(env.ccc_environment,
-                                                 params['comma_separated_queue_names'],
-                                                 params['worker_num'])
-            _rebuild_supervisor_conf_file(
-                'make_supervisor_conf',
-                'supervisor_celery_worker.conf',
-                {'celery_params': params,
-                 'worker_name': worker_name,
-                 },
-                conf_destination_filename,
-            )
-
-        if 'celery_periodic' in queue_names:
-            _rebuild_supervisor_conf_file(
-                'make_supervisor_conf',
-                'supervisor_celery_beat.conf',
-                {'celery_params': params}
-            )
-
-
-def show_periodic_server_whitelist_message_and_abort(env):
```

Comment:
I dropped this warning. I think we can decouple periodic queue and celerybeat which would solve this issue

Reply:
We could also do this validation when we validate the environment settings.

Reply:
I'm also in favor of keeping it, and validating it when we validate the environment settings. I think we still get value out of this.

Reply:
I did add https://github.com/dimagi/commcare-cloud/pull/1791/commits/bf2b8fa9032781533e28361e2fb45a2760ef0b74  in my push last night, which does validate there's only one (the one for periodic queue was already there) . are you saying we want to specify what server is hosting beat in two places so that even if you change one, we still warn if they don't match (in the hopes of catching something)?